src/store/media-store.ts

```diff
@@ -222,11 +230,66 @@ export const createSettersSlice: StateCreator<
 
 	_handleProgress: (currentTime: number) =>
 		set(state => {
+			let shouldRunTimeAlarm = false;
```

Comment:
Improve naming: `alarmTurnedOn` (You can also use sth else from the world of "alarm clocks". I found it helpful to check Wiki for naming: https://en.wikipedia.org/wiki/Alarm_clock)

Reply:
This variable could be scoped down -> see below